### PR TITLE
Feat/finish service

### PR DESCRIPTION
**Pull request by nettashviro**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# Changelog
+
+All notable changes to this project will be documented in this file.
+
+The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
+and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
+
+## [v1.0.0] 
+
+## [Unreleased]
```
Comment (nettashviro) [inline CHANGELOG.md]: add versions diff

File: proto/dropbox/dropbox.proto [hunk 2]
```diff
@@ -60,14 +73,19 @@ message HasTransferResponse {
 
 message GetTransfersInfoRequest {
   string fileID = 1;
-  string userID = 2;
+  string userID = 2; // The source user who create the request
```
Comment (nettashviro) [inline proto/dropbox/dropbox.proto]:
```suggestion
  string userID = 2; // The source user who created the request
```

File: src/dropbox/dropbox.methods.ts [hunk 3]
```diff
@@ -1,88 +1,129 @@
...
+
+    const canApprove: ICanApproveToUser = await approvalService.canApproveToUser(approverID, userID, destination);
+    return canApprove;
+  }
+
+  static async CreateRequest(call: grpc.ServerUnaryCall<any>) {
+    const params = call.request;
+    const approvers = call.request.approvers;
+
+    if (!(params.destination in Destination)) throw new ClientError(`destination value: ${params.destination}, is not supported`);
```
Comment (Shahar-Y) [inline src/dropbox/dropbox.methods.ts]: check undefined

File: src/dropbox/dropbox.methods.ts [hunk 4]
```diff
@@ -1,88 +1,129 @@
...
+      fileId: params.fileID,
+      fileName: params.fileName,
+      to: params.users,
+      from: params.sharerID,
+      info: params.info,
+      classification: params.classification,
+    };
+    await approvalService.createRequest(request, params.destination);
+
+    return {};
```
Comment (Shahar-Y) [inline src/dropbox/dropbox.methods.ts:212]:
```suggestion
    return transfer;
```

**Review by Shahar-Y — CHANGES_REQUESTED** [review]

File: proto/dropbox/dropbox.proto (hunk 2 above)
Comment (nettashviro) [reply proto/dropbox/dropbox.proto]: done

File: src/dropbox/dropbox.methods.ts (hunk 3 above)
Comment (nettashviro) [reply src/dropbox/dropbox.methods.ts]: worked

File: CHANGELOG.md (hunk 1 above)
Comment (nettashviro) [reply CHANGELOG.md]: done

File: src/dropbox/dropbox.methods.ts (hunk 4 above)
Comment (nettashviro) [reply src/dropbox/dropbox.methods.ts:212]: done

File: src/approval/approval.service.ts [hunk 5]
```diff
@@ -1,101 +1,111 @@
...
-            return res.data;
-        } catch (err) {
-            await TransferRepository.deleteByID(data.id || "");
-            if (!err.response || !err.response.status) throw new ApplicationError();
+  /**
+   * Create new transfer request in the approval service
+   * @param data - reuqest data
+   * @param destination - approval destination system
+   */
+  async createRequest(data: IRequest, destination: Destination) {
```
Comment (Shahar-Y) [inline src/approval/approval.service.ts:23]: Add @returns, even if the type is dependant on dropbox. Does it return the new request? 
If there is a type, add it to the function signature.

File: src/dropbox/dropbox.methods.ts [hunk 6]
```diff
@@ -1,88 +1,153 @@
...
-        return approverInfo;
-    }
-
-    static async CanApproveToUser(call: grpc.ServerUnaryCall<any>): Promise<ICanApproveToUser> {
-        const userID: string = call.request.userID;
-        const approverID: string = call.request.approverID;
-        const canApprove: ICanApproveToUser = await approvalService.canApproveToUser(approverID, userID);
-
-        return canApprove;
+  static async GetTransfersInfo(call: grpc.ServerUnaryCall<any>): Promise<{ transfersInfo: ITransferInfo[] }> {
```
Comment (Shahar-Y) [inline src/dropbox/dropbox.methods.ts]: Add documentation

File: src/dropbox/dropbox.methods.ts [hunk 7]
```diff
@@ -1,88 +1,153 @@
...
+          to: destUsers,
+          status: statusRes.status || transfer.status || '???',
+        };
+      })
+    );
+
+    return { transfersInfo };
+  }
+
+  static async HasTransfer(call: grpc.ServerUnaryCall<any>): Promise<{ hasTransfer: boolean }> {
```
Comment (Shahar-Y) [inline src/dropbox/dropbox.methods.ts:111]: Add documentation

File: src/dropbox/dropbox.methods.ts [hunk 8]
```diff
@@ -1,88 +1,153 @@
...
+    const userID: string = call.request.userID;
+    const fileID: string = call.request.fileID;
+
+    const hasTransfer: boolean = await TransferRepository.exists({ userID, fileID });
+
+    return { hasTransfer };
+  }
+
+  static async GetApproverInfo(call: grpc.ServerUnaryCall<any>): Promise<IApproverInfo> {
```
Comment (Shahar-Y) [inline src/dropbox/dropbox.methods.ts:126]: Add documentation

File: src/dropbox/dropbox.methods.ts [hunk 9]
```diff
@@ -1,88 +1,153 @@
...
+    const id: string = call.request.id;
+    const destination: string = call.request.destination;
+
+    if (!(destination in Destination)) throw new ClientError(`destination value: ${destination}, is not supported`);
+
+    const approverInfo: IApproverInfo = await approvalService.getApproverInfo(id, destination);
+    return approverInfo;
+  }
+
+  static async CanApproveToUser(call: grpc.ServerUnaryCall<any>): Promise<ICanApproveToUser> {
```
Comment (Shahar-Y) [inline src/dropbox/dropbox.methods.ts:143]: Add documentation

File: src/dropbox/dropbox.methods.ts [hunk 10]
```diff
@@ -1,88 +1,153 @@
...
+    const hasTransfer: boolean = await TransferRepository.exists({ userID, fileID });
+
+    return { hasTransfer };
+  }
+
+  static async GetApproverInfo(call: grpc.ServerUnaryCall<any>): Promise<IApproverInfo> {
+    const id: string = call.request.id;
+    const destination: string = call.request.destination;
+
+    if (!(destination in Destination)) throw new ClientError(`destination value: ${destination}, is not supported`);
```
Comment (Shahar-Y) [inline src/dropbox/dropbox.methods.ts:130]: nice :+1:

File: src/dropbox/dropbox.methods.ts [hunk 11]
```diff
@@ -1,88 +1,153 @@
...
+    const approverID: string = call.request.approverID;
+    const destination: string = call.request.destination;
+
+    if (!(destination in Destination)) throw new ClientError(`destination value: ${destination}, is not supported`);
+
+    const canApprove: ICanApproveToUser = await approvalService.canApproveToUser(approverID, userID, destination);
+    return canApprove;
+  }
+
+  static async CreateRequest(call: grpc.ServerUnaryCall<any>) {
```
Comment (Shahar-Y) [inline src/dropbox/dropbox.methods.ts:157]: Add documentation

File: src/dropbox/dropbox.methods.ts [hunk 12]
```diff
@@ -1,88 +1,153 @@
...
+    const params = call.request;
+    const approvers = params.approvers;
+    const destUsers = params.users;
+    const sharerID = params.sharerID;
+    const destination = params.destination;
```
Comment (Shahar-Y) [inline src/dropbox/dropbox.methods.ts]: Add types

File: src/status/status.interface.ts [hunk 13]
```diff
@@ -1,6 +1,14 @@
...
+export type Status = {
+  status: string,
+  displayName: string,
+};
 export interface IStatus {
-    users: IUser[];
-    status: string;
-}
\ No newline at end of file
+  id: string;
```
Comment (Shahar-Y) [inline src/status/status.interface.ts]: id is the reqID? if so, add it as a comment.
* Actually add either way :sweat_smile:

File: src/utils/logger.ts [hunk 14]
```diff
@@ -1,6 +1,37 @@
...
+  indexPrefix: confLogger.indexPrefix,
+  level: 'verbose',
+  clientOpts: confLogger.options,
+  bufferLimit: 100,
+  messageType: 'log',
+  ensureMappingTemplate: true,
+  mappingTemplate: indexTemplateMapping,
+};
+const elasticsearch: WinstonElasticsearch.default = new Elasticsearch(options);
+logger.add(elasticsearch);
```
Comment (Shahar-Y) [inline src/utils/logger.ts:30]: Amazing! Best code EVER!

**Review by Shahar-Y — COMMENTED** [review]
Found very little comments, mostly about lack of documentation in some functions. 
Great job, let's make it work! :muscle:

File: src/approval/approval.service.ts (hunk 5 above)
Comment (nettashviro) [reply src/approval/approval.service.ts:23]: returns an empty object

File: src/dropbox/dropbox.methods.ts (hunk 6 above)
Comment (nettashviro) [reply src/dropbox/dropbox.methods.ts]: done

File: src/dropbox/dropbox.methods.ts (hunk 7 above)
Comment (nettashviro) [reply src/dropbox/dropbox.methods.ts:111]: done

File: src/dropbox/dropbox.methods.ts (hunk 8 above)
Comment (nettashviro) [reply src/dropbox/dropbox.methods.ts:126]: done

File: src/dropbox/dropbox.methods.ts (hunk 9 above)
Comment (nettashviro) [reply src/dropbox/dropbox.methods.ts:143]: done

File: src/dropbox/dropbox.methods.ts (hunk 11 above)
Comment (nettashviro) [reply src/dropbox/dropbox.methods.ts:157]: done

File: src/dropbox/dropbox.methods.ts (hunk 12 above)
Comment (nettashviro) [reply src/dropbox/dropbox.methods.ts]: done

File: src/status/status.interface.ts (hunk 13 above)
Comment (nettashviro) [reply src/status/status.interface.ts]: done

File: src/utils/logger.ts (hunk 14 above)
Comment (nettashviro) [reply src/utils/logger.ts:30]: :satisfied: :woman_facepalming:

**Review by Shahar-Y — APPROVED** [review]
Great work!

**Pull request merged**